github/Requester.py

```diff
@@ -815,7 +815,7 @@ def createException(
             exc = GithubException.BadUserAgentException
         elif status == 403 and cls.isRateLimitError(message):
             exc = GithubException.RateLimitExceededException
-        elif status == 404 and message == "not found":
+        elif status == 404 and (message == "not found" or "no object found" in message):
```

Comment:
```suggestion
        elif status == 404 and (message == "not found" or "no object found" in message.lower()):
```

It will fail for the cases when message is `"No object found for the path <path>"`

Reply:
message is lowercase already, see above, the test uses "No object found ..." and passes.